utils/import_data.py

```diff
@@ -52,161 +52,161 @@ def import_data(fw, df, dry_run=False):
+                objects_for_processing = fu.get_objects_for_processing(
+                    fw, ses, "acquisition", get_files
                 )
-                continue
 
-            # Names must exist, so warn if nothing maches
-            elif len(matches) == 0:
-                log.warning(f"No match for object name '{object_name}'.")
                 log.info(
-                    "\n--------------------------------------------------\n"
-                    "STATUS: Failed\n"
-                    "==================================================\n"
+                    f"\n==================================================\n"
+                    f"Setting Metadata For {object_name}\n"
+                    f"--------------------------------------------------\n"
                 )
-                continue
-
-            # Take the (hopefully) one match
-            match = matches[0]
+                log.info(series)
 
-            # Generate a flywheel path to this file.  A little redundant since we
-            # already have group/project/subject/session, but this will also find
-            # acquisition and file.
-            address = fh.generate_path_to_container(fw, match)
+                # Find any files that match the name of the file specified in this row of the dataframe
+                matches = [
+                    m for m in objects_for_processing if m.get("name") == object_name
+                ]
 
-            # We write ROI info to the parent Session metadata, so get the containing session
-            # of the matching file.
-            parent = match.parent
-            if parent.container_type == "session":
-                ses = parent.reload()
-
-            else:
-                ses = fw.get_session(parent.parents.session)
-
-            # Get an ROI object from the row using required and optional columns.
-            roi = cu.get_roi_from_row(series, match, ses)
+                if len(matches) == 0:
+                    log.debug(
+                        f"No matches found for {object_name}, appending File type '.{series.get('file type')}'"
+                    )
+                    object_name += f".{series.get('File Type')}"
 
-            # If this gear is a dry run, we'll only log, not actually upload
-            if dry_run:
-                log.info(f"Would modify info on {address}")
-                df.loc[df.index == row, "Gear_Status"] = "Dry-Run Success"
-                df.loc[df.index == row, "Gear_FW_Location"] = address
-                log.info(
-                    "\n--------------------------------------------------\n"
-                    "DRYRUN STATUS: Success\n"
-                    "==================================================\n"
-                )
-                success_counter += 1
+                log.debug(f"looking for {object_name}")
 
-            # Otherwise make the ROI
-            else:
+                matches = [
```

Comment:
So here what you're doing is looking for files in acquisition containers under this session that match the name in the CSV, and then if there are no matches you look files that match `{object_name}.{file_type}`?  Could you look for files under that session that contain `object_name` instead of having to loop through twice?

Reply:
It's kind of like a "In case you forgot the extensions" last ditch attempt.  Fun fact this was put in to catch things I saw going on at imidex.....